offchain-modules/packages/app-multisign-server/src/ckbSigner.ts

```diff
@@ -185,13 +206,73 @@ async function verifyMintTx(
 }
 
 async function verifyUnlockTx(
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   pubKey: string,
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   rawData: string,
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   payload: ckbUnlockCollectSignaturesPayload,
 ): Promise<SigError> {
+  const collectorScript = parseAddress(getFromAddr());
+  const omniLockMultisigScript = parseAddress(getOmniLockMultisigAddress());
+  const txSkeleton = payload.txSkeleton;
+  const sigData = txSkeleton.signingEntries[1].message;
+  if (rawData !== sigData) {
+    return new SigError(SigErrorCode.InvalidParams, `rawData:${rawData} doesn't match with:${sigData}`);
+  }
+
+  const unlockRecords = payload.unlockRecords;
+  asserts(unlockRecords);
+
+  let err = await verifyDuplicateUnlockTx(pubKey, unlockRecords, txSkeleton);
+  if (err.Code !== SigErrorCode.Ok) {
+    return err;
+  }
+
+  const sudtTypescript = ForceBridgeCore.config.ckb.deps.sudtType.script;
+  if (
+    txSkeleton.outputs.some(
+      (output) => output.cell_output.type && output.cell_output.type.code_hash !== sudtTypescript.codeHash,
+    )
+  ) {
+    return new SigError(SigErrorCode.InvalidParams, `unlock recode outputs contains non-sudt typescript`);
+  }
+  const unlockCells = txSkeleton.outputs.filter(
+    (cell) =>
+      cell.cell_output.lock.code_hash !== collectorScript.code_hash &&
+      cell.cell_output.lock.code_hash !== omniLockMultisigScript.code_hash,
+  );
+  if (unlockRecords.length !== unlockCells.length) {
+    return new SigError(
+      SigErrorCode.InvalidParams,
+      `unlok recode length:${unlockRecords.length} doesn't match with:${unlockCells.length}`,
+    );
+  }
+
+  const typeArgs = unlockCells.map((output) => {
+    const typescript = output.cell_output.type;
+    return typescript ? typescript.args : '';
+  });
+
+  if (typeArgs.length > 1) {
+    return new SigError(SigErrorCode.InvalidParams, `unlock recode outputs contains not single typescript`);
+  }
```

Comment:
It seems typeArgs.length === unlockCells.length ?

Reply:
fixed to verify distinct(hash(typescript)).length > 1